pkg/config/process_test.go

```diff
@@ -195,3 +221,77 @@ func TestProcBindEnvAndSetDefault(t *testing.T) {
 	// Make sure the default is set properly
 	assert.Equal(t, "asdf", cfg.GetString("process_config.foo.bar"))
 }
+
+func TestProcBindEnv(t *testing.T) {
+	cfg := setupConf()
+	procBindEnv(cfg, "process_config.foo.bar")
+
+	envs := map[string]struct{}{}
+	for _, env := range cfg.GetEnvVars() {
+		envs[env] = struct{}{}
+	}
+
+	_, ok := envs["DD_PROCESS_CONFIG_FOO_BAR"]
+	assert.True(t, ok)
+
+	_, ok = envs["DD_PROCESS_AGENT_FOO_BAR"]
+	assert.True(t, ok)
+
+	// Make sure that DD_PROCESS_CONFIG_FOO_BAR shows up as unset by default
+	assert.False(t, cfg.IsSet("process_config.foo.bar"))
+
+	// Try and set DD_PROCESS_CONFIG_FOO_BAR and make sure it shows up in the config
+	reset := setEnvForTest("DD_PROCESS_CONFIG_FOO_BAR", "baz")
+	assert.True(t, cfg.IsSet("process_config.foo.bar"))
+	assert.Equal(t, "baz", cfg.GetString("process_config.foo.bar"))
+	reset()
+}
+
+func TestDisplayProcConfigEnabledDeprecationWarning(t *testing.T) {
+	assert.Error(t, displayProcConfigEnabledDeprecationWarning())
+	assert.NoError(t, displayProcConfigEnabledDeprecationWarning()) // Make sure warning only shows once
+}
+
+func TestGetProcessAndContainerCollectionEnabled(t *testing.T) {
+	for _, tc := range []struct {
+		name                                             string
+		procConfigEnabled                                string
+		containerCollection, containerCollectionExpected bool
+		processCollection, processCollectionExpected     bool
+	}{
+		{
+			name:                        "process_config.enabled set",
+			procConfigEnabled:           "true",
```

Comment:
Should we also test the cases with `false` and `disabled` to make sure that they're correctly mapped to the new settings?

Reply:
Same here, I'll add a test case for this